Wikipedia/Code/UserHistoryFunnel.swift

```diff
@@ -95,10 +96,14 @@ private typealias ContentGroupKindAndLoggingCode = (kind: WMFContentGroupKind, l
         guard isTarget else {
             return
         }
-        
+        guard let lastAppVersion = UserDefaults.wmf_userDefaults().wmf_lastAppVersion else {
+            assertionFailure("Last app version should be set by now")
```

Comment:
could the initial set of `lastAppVersion` be moved from `AppDelegate` to here to keep it contained within one class?

Reply:
Not really, User history snapshot is logged at the end of the session so if we set `lastAppVersion` at that time, it will be the same as current `appVersion`. I can wrap it in `setLastAppVersion` method on `UserHistoryFunnel` and call it in `AppDelegate` if you think that's better

Reply:
but we could set it in `logStartingSnapshot`

Reply:
I mean just in the `nil` case - rather than having an assertion here, you could set the initial value here (you could also potentially use this to log the initial upgrade from 5.8 to 6.0 by logging when the previous value was nil)

Reply:
we're gonna log the first snapshot no matter what in `logStartingSnapshot ` and then once we want to log another one, we should be good setting `lastAppVersion` in `logSnapshot` since it's only necessary for that check in there, no? 🤔

Reply:
If you `logStartingSnapshot` without setting the version, then log on `nil` version in `logSnapshot`, won't it log twice for new users?  Setting the version in `logStartingSnapshot` when it reaches `log(event())` should prevent the double log

Reply:
yeah that's a good point 🤯, but we should be ok cause we're setting `wmf_lastAppVersion` to `WikipediaAppUtils.appVersion()` in `logged` so after the first log, `wmf_lastAppVersion` won't be nil